### PR TITLE
fix(partition): allow custom sorting for the legend items

### LINKED ISSUE #1935: [Partitions] Use sorting predicate prop to sort legend items
**Describe the issue**
The `Settings` component exposes a `legendSort` predicate prop to sort legend items in the visualization, but the Partition charts are not honouring it.

**To Reproduce**
Check on this codesandbox the console deck: no item is printed.

[Edit misty-flower-emtrld](https://codesandbox.io/s/misty-flower-emtrld?fontsize=14&hidenavigation=1&theme=dark)

**Kibana Cross Issues**
[Linked issue](https://github.com/elastic/kibana/issues/149034)

**Checklist**

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] Every related Kibana issue is listed under `Kibana Cross Issues` list and the `kibana cross issue` tag is applied

### PR DESCRIPTION
**Pull request by dej611**
## Summary

The `legendSort` prop is now applied to partition charts allowing the user to custom sort (flattened) legend items.

partition_custom_legend_sort

## Issues

fix #1935 

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [ ] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated

### CODE REVIEW

**Comment by markov00** [comment]
@dej611 let me know if you need any help writing some visual tests for this changes

**Comment by dej611** [comment]
> @dej611 let me know if you need any help writing some visual tests for this changes

That would be helpful

File: storybook/stories/legend/10_sunburst.story.tsx [hunk 1]
```diff
@@ -50,6 +53,26 @@ export const Example = () => {
   });
   const legendStrategy = select('legendStrategy', LegendStrategy, LegendStrategy.Key as LegendStrategy);
   const maxLines = number('max legend label lines', 1, { min: 0, step: 1 });
+
+  const legendSortStrategy = select(
+    'Custom legend sorting',
+    { RegionsFirst: 'regionsFirst', ProductsFirst: 'productsFirst', DefaultSort: 'default' },
+    'default',
```
Comment (dej611) [inline storybook/stories/legend/10_sunburst.story.tsx:60]:
```suggestion
    'regionsFirst',
```

**Comment by dej611** [comment]
buildkite update screenshots

**Pull request merged**